smart-client-core/src/main/java/com/emc/rest/smart/SmartClientException.java

```diff
@@ -0,0 +1,28 @@
+package com.emc.rest.smart;
+
+public class SmartClientException extends RuntimeException {
+
+    private ErrorType errorType = ErrorType.Unknown;
+
+    public SmartClientException(String message) {
+        super(message);
+    }
+
+    public SmartClientException(String message, Throwable cause) {
+        super(message, cause);
+    }
+    public ErrorType getErrorType() {
+        return errorType;
+    }
+
+    public void setErrorType(ErrorType errorType) {
+        this.errorType = errorType;
+    }
+
```

Comment:
can we add some convenience methods here (`isClientError()` and `isServerError()`)